src/dex.rs

```diff
@@ -772,22 +772,73 @@ impl DexReader {
             inner,
         })
     }
+
+    /// Loads a `Dex` from a `Vec<u8>`
+    pub fn from_vec(vec: Vec<u8>) -> Result<Dex<Vec<u8>>> {
```

Comment:
Is a `AsRef<[u8]>` or an `[u8]` better here instead of taking ownership of the vec?

Reply:
Both `AsRef<[u8]>` and `[u8]` do not have a known size at compilation time, which seems to be a requirement of the type parameter `T` in `Dex<T>`. Since `Vec` wraps the buffer with a fixed structure it does not have that problem. So, unless I'm missing something, I think a `Vec` is the way to go.

Reply:
`[u8]` is dynamically sized but `&[u8]` is fixed-size. Also, the function could have a signature of `(impl AsRef<[u8]>) -> Result<Dex<&[u8]>>`.

Reply:
I tried @sollyucko's suggestion but `impl AsRef<[u8]>` does not implement `pread`, which is, if I'm not mistaken, the entry point to the parsing library used under the hood.

This function signature, however, compiles and passes the tests:

```rust
pub fn from_ref<'a>(vec: &'a [u8]) -> Result<Dex<&'a [u8]>>
```

So, if @letmutx likes it, I can update the PR that way.

Reply:
`AsRef<[u8]>` gives more flexibility. You have to deref before using `pread`
```rust
 pub fn from_vec<B: AsRef<[u8]>>(buf: B) -> Result<Dex<B>> {
        let inner: DexInner = buf.as_ref().pread(0)?;
        ...
}
```

Reply:
Ok, I changed the signature and the test passes, so I think is good to go. I don't think I had `AsRef<>` 100% figured out and that caused some confusion on my side.